scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -455,7 +455,41 @@ static s32 pick_idle_cpu(struct task_struct *p, struct task_ctx *taskc,
 			goto found_cpu;
 		}
 
-		// TODO: handle affinitized tasks to be more topology aware
+		// First try to find an idle SMT in the LLC
+		if (smt_enabled && taskc->mask && llcx->cpumask) {
+			bpf_cpumask_and(taskc->mask, cast_mask(llcx->cpumask),
+					p->cpus_ptr);
+			if (taskc->mask &&
+			    (cpu = scx_bpf_pick_idle_cpu(cast_mask(taskc->mask),
+							 SCX_PICK_IDLE_CORE)) >= 0) {
+				*is_idle = true;
+				goto found_cpu;
+			}
+		}
+
+		// Next try to find an idle CPU in the LLC
+		if (taskc->mask && llcx->cpumask) {
+			bpf_cpumask_and(taskc->mask, cast_mask(llcx->cpumask),
```

Comment:
taskc->mask seems the same as the smt_enabled case, you don't need to re-compute it, maybe you can move it all the way up at the beginning.